src/main/scala/encry/network/EncryDeliveryManager.scala

```diff
@@ -65,7 +79,9 @@ class EncryDeliveryManager(syncInfoSpec: EncrySyncInfoMessageSpec.type) extends
       historyReaderOpt.foreach { h =>
         val currentQueue: Iterable[ModifierId] = cancellables.keys.map(ModifierId @@ _.toArray)
         val newIds: Seq[(ModifierTypeId, ModifierId)] = h.modifiersToDownload(settings.network.networkChunkSize - currentQueue.size, currentQueue)
-        newIds.groupBy(_._1).foreach(ids => requestDownload(ids._1, ids._2.map(_._2)))
+          .filter(modId => !cancellables.keys.toSet.contains(modId._2))
+        if (newIds.nonEmpty) newIds.groupBy(_._1).foreach(ids => requestDownload(ids._1, ids._2.map(_._2)))
```

Comment:
You can use smth like `case(modifierId, ids) => requestDownload(modifierId, ids)` inside foreach, it's more readable